libcasr/src/ubsan.rs

```diff
@@ -0,0 +1,151 @@
+//! UndefinedBehaviorSanitizer module implements `Exception` traits for UndefinedBehaviorSanitizer reports.
+use crate::exception::Exception;
+
+use crate::error::*;
+use crate::execution_class::ExecutionClass;
+use regex::Regex;
+
+/// Structure provides an interface for parsing ubsan runtime error message.
+pub struct UbsanException;
```

Comment:
I think, we should have a struct that some fields:
```Rust
pub struct UbsanException { // or Find some better name UbsanWaring for e.g.
        pub warning: String,
        pub class: Option<ExecutionClass>.
        pub stacktrace: Option<String>,
}
type UbsanWarings =  Vec<UbsanWarnings>;
```
In such way we don't  function `extract_ubsan_reports`.  Maybe we could create  1 method that returns a `UbsanWarings`.

Reply:
Or maybe we should have 2 methods, that creates UbsanWarning without class and a method  in ubsan warning that updates itself and gets the class.